provisioning.adoc

```diff
@@ -0,0 +1,303 @@
+version of the storage system. Limiting the scope of the Internal Provisioning
+Scenario allows specially targeted deployments. It also limits the ssh access
+between the nodes of a single cluster. Any changes or updates to the
+Provisioner will also only affect a specific cluster instance, including the
+failure and re-election of the Provisioning Controller.
+
+
+=== External Execution Scenario
+
+TODO: Discussion of the expansion of the node agent deployment on all the nodes,
+including the tendrl api, etcd, monitoring stack etc. This means that any of
+these nodes can function as the Provisioning Control Node, in addition to being
+monitored and provisioned by the node agent.
+
+
+== Workflows
+
+=== Create Cluster
+
+==== What Needs to be Provisioned by Tendrl
+* Storage system (ceph, gluster)
+* Tendrl-sds bridge (specific to the storage system)
+* Layered storage system (nfs-ganesha, samba + ctdb)
+
+==== Flow
+. Tendrl core deployment is up and running with all the Provisioner
+Provisioning Modules installed.
+. Parameters required for provisioning a specific version of the storage system
+are known to tendrl via the YAML definition made available by the Provisioner
+Provisioning Modules.
+. Tendrl API exposes the endpoint to accept a provisioning task with these
+parameters based on the provisioning workflow that pulls in these YAML
+definition files.
+. As part of the required parameters for a create cluster flow, a list of all
+the nodes which will form a part of that cluster needs to be provided. Tendrl
+could internally validate that the node agent is indeed running on all of these
+nodes as expected.
+. The create cluster call also requires the type of storage system and a
+specific version to be deployed. Tendrl can internally validate the support for
+this type-version combination based on the Provisioner Provisioning YAML file
+definitions.
+. Upon receiving the correct create cluster call, tendrl (node agents or
+whoever is eventually deemed to be responsible) designates a Provisioning
+Control Node and the Provisioning Controller.
+. Provisioning Controller invokes the Provisioner Interface Modules to setup
+the Provisioner.
+. Provisioning Controller also invokes and waits upon the successful outcome of
+user account creation via the other node agents that reside on other nodes that
+are to be part of the cluster.
+. Provisioning Controller invokes `ansible ping’ or equivalent to validate the
+connectivity between the Provisioning Control Node and the other nodes in the
+cluster.
+. Provisioning Controller invokes the Provisioner Interface Modules to invoke
+the Provisioner to setup the cluster.
+. Provisioning controller invokes the Native Execution Scenario on the specific
+nodes to deploy the bridge(s) via the central store.
+
+
+== Import Cluster
+
+==== What Needs to be Provisioned by Tendrl
+* Tendrl-sds bridge (specific to the storage system)
+
+==== Flow
```

Comment:
I think of a scenario where we have two gluster clusters having 10 nodes each. On with say gluster-3.9.2 and other set with gluster-3.8.x.
All these nodes are known to tendrl once tendrl node-agent is up and running on these nodes. How does the api layer or UI layer decide what nodes to be passed for each of the two imports?

I feel this flow should be described with more clarity here.